lib/src/folding/parsers/spaces_foldable_block_parser.dart

```diff
@@ -0,0 +1,109 @@
+import 'dart:collection';
+
+import 'package:charcode/ascii.dart';
+import '../foldable_block.dart';
+import '../foldable_block_type.dart';
+
+import 'abstract.dart';
+
+///A parser for foldable blocks from spaces count at start of each line.
+class SpacesFoldableBlockParser extends AbstractFoldableBlockParser {
+  static const _kSeparatorLine = -1;
+
+  void parse(String text) {
+    _addWhitespacesBlocks(text);
+    finalize();
+  }
+
+  void _addWhitespacesBlocks(String text) {
+    final spacesCountBeforeLines = _countWhitespacesInStartOfEachLine(text);
+
+    final openBlocksBySpacesCount = HashMap<int, List<FoldableBlock>>();
+    final resultBlocks = List<FoldableBlock>.empty(growable: true);
+
+    int lastValuableLine = spacesCountBeforeLines.length - 1;
+
+    for (int i = 0; i < spacesCountBeforeLines.length - 1; i++) {
+      final currentSpacesCount = spacesCountBeforeLines[i];
+
+      if (currentSpacesCount == _kSeparatorLine) {
+        continue;
+      }
+
+      final nextExistingSpacesCount =
+          spacesCountBeforeLines.skip(i + 1).firstWhere(
+                (element) => element != _kSeparatorLine,
+                orElse: () => _kSeparatorLine,
+              );
+
+      if (nextExistingSpacesCount == _kSeparatorLine) {
+        lastValuableLine = i;
+        break;
+      }
+
+      if (nextExistingSpacesCount > currentSpacesCount) {
+        if (openBlocksBySpacesCount[currentSpacesCount] == null) {
+          openBlocksBySpacesCount[currentSpacesCount] =
+              List.empty(growable: true);
+        }
+
+        openBlocksBySpacesCount[currentSpacesCount]!.add(
+          FoldableBlock(
```

Comment:
Do not reuse this class just because you need 1 of its 3 fields. One of the many reasons is that we may add fields there, and so this place has one more alien reason to change.

The parent solves this with `_StartedFoldableBlock` class.

Reply:
used just int value instead of creating new class